client/src/App.vue

```diff
@@ -0,0 +1,177 @@
+    <button type="button" class="btn btn-info action_btn" v-on:click="downloadCSVData">
+      Download
+</button>
+  </div>
+</template>
+
+<script>
+
+
+import axios from 'axios'
+
+export default {
+  name: "App",
+  data() {
+    return {
+        searchQuery: null,
+        stores:[]
+    };
+  },
+  beforeCreate(){
+
+    var self = this
+    const storesUrl = 'http://localhost:8090/stores'
+    const saveStoreDataInDatabaseUrl = "http://localhost:8090/saveStoreDataInDatabase"
+    const saveStoreAndSeasonDataInDatabaseurl = "http://localhost:8090/saveStoreAndSeasonDataInDatabase"
+
+      axios.get(saveStoreDataInDatabaseUrl, {
+      dataType: 'json',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      },
+      mode: 'no-cors',
+      credentials: 'include'
+    })
+    .then(function (response) {
+      console.log(JSON.stringify(response.data))
+    })
+    .catch(function (error) {
+      console.log(error)
+    })
+
+      axios.get(saveStoreAndSeasonDataInDatabaseurl, {
+      dataType: 'json',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      },
+      mode: 'no-cors',
+      credentials: 'include'
+    })
+    .then(function (response) {
+      console.log(JSON.stringify(response.data))
+    })
+    .catch(function (error) {
+      console.log(error)
+    })
+
+    axios.get(storesUrl, {
+      dataType: 'json',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      },
+      mode: 'no-cors',
+      credentials: 'include'
+    })
+    .then(function (response) {
+      console.log(JSON.stringify(response.data))
+      self.stores = response.data
+    })
+    .catch(function (error) {
+      console.log(error)
+    })
+
+  },
+  computed: {
+    resultQuery(){
+      if(this.searchQuery){
+      return this.stores.filter((store)=>{
+        
+        console.log(JSON.stringify(store))
+        return this.searchQuery.toLowerCase().split(' ').every(v => store.name?.toLowerCase().includes(v))
+      })
+      }else{
+        return this.stores;
+      }
+    }
+  },
+  methods: {
+    saveStore(store){
+      const saveUpdatedStoreUrl = 'http://localhost:8090/store/'
+      console.log(JSON.stringify(store))
+      axios.put(saveUpdatedStoreUrl + store.id, store,{
+      dataType: 'json',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      mode: 'no-cors',
+      credentials: 'include'
+    })
+    .then(function (response) {
+      console.log(JSON.stringify(response.data))
+    })
+    .catch(function (error) {
+      console.log(error)
+    })
+    },
+    downloadCSVData() {
+      let csv = 'id,code,description,name,openingDate,storeType,season\n';
+
+      let arr = [];
+      this.stores.forEach(store => {
+        arr.push(Object.values(store))
+      })
+    
+      arr.forEach((row) => {
+              csv += row.join(',');
```

Comment:
Hello @jaypee01 thank you for your submission! :)

I am a member of the team evaluating your challenge, and I wanted to ask you some questions.

This function has a limitation because it does not comply with the CSV standard in totality.

Can you tell me what this limitation might be?

Also, in order to avoid re-implementing a CSV printer (which paves the way for human-error), why didn't you use an existing library to generate this CSV?

Reply:
Thanks for reviewing.

In understand, It will need modification, if csv columns name and their count changes.

This should be generic , so that it could be used for any csv export (based upon data provided).

I have not used existing library, Due to assignment submission time constraint, I could not do that.

Reply:
Ok. Yes, this that you mention is indeed a design limitation. But we understand that time is a constraint, so no problem with that.

The issue regarding the CSV standard I was mentioning is the following: In a row, if you receive a column that contains a `,` character, will it collide with the delimiter `,` character in the CSV? How are you handling this?

The advantage with an external lib is that it handles these corner cases automatically for you.

Reply:
Yes it can collide. To avoid that,  we can quote the column value , then it wont consider that "," as delimiter.